### PR TITLE
Remove obsolete class RcLinkedListImpl

### LINKED ISSUE #3210: Obsolete class RcLinkedListImpl can be removed
We do not use field RcLinkedListImpl::_isOrphan to check for orphan classes in the shared cache anymore. (see https://github.com/eclipse/openj9/pull/3199). Thus the class RcLinkedListImpl can be removed.

### PR DESCRIPTION
**Pull request by hangshao0**
Fixes #3210

Signed-off-by: hangshao <hangshao@ca.ibm.com>

### CODE REVIEW

File: runtime/shared_common/ROMClassManagerImpl.hpp [hunk 1]
```diff
@@ -104,19 +103,13 @@ class SH_ROMClassManagerImpl : public SH_ROMClassManager
 	 *
 	 * Local hashtable (_rcTable) has an entry for each known ROMClass name.
 	 * Key is the fully-qualified string name of the ROMClass
-	 * Value is a RcLinkedListImpl list of ROMClasses stored under that name
-	 * A RcLinkedListImpl is a reference to a ROMClass in the cache.
+	 * Value is a HashLinkedListImpl list of ROMClasses stored under that name
```
Comment (pshipton) [inline runtime/shared_common/ROMClassManagerImpl.hpp]: This whole comment seems strangely placed.

**Comment by pshipton** [comment]
jenkins test all win jdk8

**Comment by pshipton** [comment]
PR tests
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK8-win_x86-64_cmprssptrs-OpenJ9/302/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK8-win_x86-64_cmprssptrs-OpenJ9/41/

File: runtime/shared_common/ROMClassManagerImpl.hpp (hunk 1 above)
Comment (hangshao0) [reply runtime/shared_common/ROMClassManagerImpl.hpp]: This is a comment for  _linkedListImplPool. I have put it before _linkedListImplPool.

**Pull request merged**